core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -30,13 +32,12 @@
  * <p>
  * This can be extended by providing a {@link TableOperations} to the constructor.
  */
-public class BaseTable implements Table, HasTableOperations {
-  private final TableOperations ops;
-  private final String name;
+public class BaseTable implements Table, HasTableOperations, Serializable {
+  private transient TableOperations ops;
+  private transient String name;
```

Comment:
These shouldn't need to change if `writeReplace` is implemented. This object won't be serialized, it will be replaced by the one to serialize.

Reply:
Okay, after reading the `StaticTable`, I see what you're doing. But I don't think that it is a good idea for this to make its fields transient. Can we use a `readResolve` method to create a `BaseTable` directly instead of having a `StaticTable`?

Reply:
I was toying with implementing the `readResolve` too. I decided against it because I thought it is more explicit to return a `StaticTable` object after deserialization instead of returning a `BaseTable` where the `ops` are `StaticTableOperations` and this way changing the behavior of the deserialized object, but hiding the details.

Solved this by adding comments.